platform/ios/src/MGLMapView.mm

```diff
@@ -3348,12 +3348,16 @@ - (MGLMapCamera *)cameraThatFitsCoordinateBounds:(MGLCoordinateBounds)bounds edg
     return [self cameraForCameraOptions:cameraOptions];
 }
 
-- (MGLMapCamera *)cameraThatFitsCoordinateBounds:(MGLCoordinateBounds)bounds direction:(CLLocationDirection)direction pitch:(CGFloat)pitch edgePadding:(UIEdgeInsets)insets
-{
+- (MGLMapCamera *)camera:(MGLMapCamera *)camera fittingShape:(MGLShape *)shape edgePadding:(UIEdgeInsets)insets {
     mbgl::EdgeInsets padding = MGLEdgeInsetsFromNSEdgeInsets(insets);
     padding += MGLEdgeInsetsFromNSEdgeInsets(self.contentInset);
-    mbgl::CameraOptions cameraOptions = _mbglMap->cameraForLatLngBounds(MGLLatLngBoundsFromCoordinateBounds(bounds), padding, direction, pitch);
-    return [self cameraForCameraOptions:cameraOptions];
+    
+    CGFloat pitch = camera.pitch;
+    CLLocationDirection direction = camera.heading;
```

Comment:
When `pitch` or `heading` is negative, it should be ignored in favor of the most appropriate value (which can be the current value).

Reply:
Hmm this brings up a good point. I feel like the lowest level function should be responsible for normalizing the direction.

Reply:
This is the lowest-level method at which the normalization is relevant. The negative-means-ignored behavior is specific to Apple platforms.